.github/workflows/test_pull_requests.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-python@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.x"
```

Comment:
why no cache here?

Reply:
just cause it's only two packages, it is never the rate limiting job.  keeping things simple if they don't need it